server/manager/src/main/java/org/apache/accumulo/manager/TabletGroupWatcher.java

```diff
@@ -1004,28 +1027,43 @@ private List<Range> createRangesForDeletion(TabletMetadata tabletMetadata,
     return ranges;
   }
 
-  private void updateMetadataRecordsForDelete(MergeInfo info) throws AccumuloException {
+  private Pair<KeyExtent,KeyExtent> updateMetadataRecordsForDelete(MergeInfo info)
+      throws AccumuloException {
     final KeyExtent range = info.getExtent();
 
     String targetSystemTable = MetadataTable.NAME;
     if (range.isMeta()) {
       targetSystemTable = RootTable.NAME;
     }
+    final Pair<KeyExtent,KeyExtent> startAndEndTablets;
 
     final AccumuloClient client = manager.getContext();
 
     try (BatchWriter bw = client.createBatchWriter(targetSystemTable)) {
       final Text startRow = range.prevEndRow();
-      final Text lastRow = range.endRow() != null
+      final Text endRow = range.endRow() != null
           ? new Key(range.endRow()).followingKey(PartialKey.ROW).getRow() : null;
 
-      // Find the tablets that overlap the start and end of the deletion range
-      // Store in a Map so that if we have the same Tablet we only need to process once
+      // Find the tablets that overlap the start and end row of the deletion range
+      // If the startRow is null then there will be an empty startTablet we don't need
+      // to fence a starting tablet as we are deleting everything up to the end tablet
+      // Likewise, if the endRow is null there will be an empty endTablet as we are deleting
+      // all tablets after the starting tablet
+      final Optional<TabletMetadata> startTablet =
+          Optional.ofNullable(startRow).flatMap(er -> loadTabletMetadata(range.tableId(), startRow,
+              ColumnType.PREV_ROW, ColumnType.FILES));
+      final Optional<TabletMetadata> endTablet = Optional.ofNullable(endRow).flatMap(
+          er -> loadTabletMetadata(range.tableId(), endRow, ColumnType.PREV_ROW, ColumnType.FILES));
```

Comment:
If the following is correct, would be a bit cleaner to use the lambda parameter.

```suggestion
      final Optional<TabletMetadata> startTablet =
          Optional.ofNullable(startRow).flatMap(row -> loadTabletMetadata(range.tableId(), row,
              ColumnType.PREV_ROW, ColumnType.FILES));
      final Optional<TabletMetadata> endTablet = Optional.ofNullable(endRow).flatMap(
          row -> loadTabletMetadata(range.tableId(), row, ColumnType.PREV_ROW, ColumnType.FILES));
```

Reply:
Thanks for catching that, that was an oversight on my part.